### DISCUSSION TITLE
How to ask a question

### DISCUSSION THREAD

**Discussion by AlanSpatial** [spatialsys/spatial-unity-sdk#12] [Announcements]
When creating a discussion asking for help it is critical that you provide enough information in your initial post for somebody to be able to help you properly. The following are some guidelines you should be thinking about when creating a new discussion.

# Use a Descriptive Title
The title of your post should accurately and _specifically_ describe your problem or question. If you are having issues with a feature, that specific feature should likely make it into the title.

**Bad Title ❌**
> Help with avatar

**Good Title ✅**
> How to play embedded emotes on avatar

# Explain Your Goal
Its often easier to address somebodies question when you understand their desired result. Often when you need help you might not even know the correct question to ask. Explaining your end goal helps others fill in the blanks and solve the root issue.

# Show Your Code
If you are having a problem with code you are writing make sure to share it! 

[Code can be pasted inside github nicely by wrapping it in tripple `s (```)
](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code)
```csharp
public float example; // :)
```

# List the Solutions You've Tried
Explain what sort of work you have already done to try and resolve the issue. Don't forget to check the Creator Toolkit documentation first.

[Creator Toolkit Documentation](https://docs.spatial.io/)

# Search Before Asking
The fastest way to get a question answered is to look to see if somebody asked it before you. Google is your friend, and since Spatial uses Unity and C# there is already a wealth of knowledge available for less Spatial-specific questions.

**Comment by iweavings** [comment]
thank you so much for setting up this post.
I must say,... I sent this link to 12 people so far.
it has been a game changer for me to learn how to formulate adequate questions for IT support on so many platforms. 
(and I'm not even good at it yet).
